client/src/components/BulkCreateForm/BulkCreateForm.js

```diff
@@ -76,164 +172,298 @@ export default class BulkCreateForm extends React.Component {
+        return (
+            <Card className="bulk-create-form-container">
+                <Card.Title>
+                    Bulk Create Monuments and Memorials
+                </Card.Title>
+                {!showFieldMapping && !showValidationResults && <>
+                    {!fileUpload.csv && !fileUpload.zip && this.renderFileUpload()}
+                    {fileUpload.images.length > 0 && this.renderUploadedFiles()}
+                </>}
+                {showFieldMapping && !showValidationResults && this.renderFieldMapping()}
+                {showValidationResults && this.renderValidationResults()}
+            </Card>
+        );
     }
 
-    /**
-     * Validates the Form
-     * If any of the inputs are invalid, the entire Form is considered invalid
-     * @returns {boolean} - True if the Form is valid, False otherwise
-     */
-    validateForm() {
+    renderFileUpload() {
+        const { fileUpload, fileUploadInputKey } = this.state;
+        return (<Card.Body>
+            <Card.Subtitle className="mt-2">
+                CSV Upload
+            </Card.Subtitle>
+            <p className="mb-4">
+                You can create multiple monuments or memorials by uploading
+                a <code>.csv</code> file with information about them.
+            </p>
+            <Card.Subtitle>
+                Zip Upload
+            </Card.Subtitle>
+            <p className="mb-4">
+                If you're uploading images with your monuments, you must create
+                a <code>.zip</code> file containing
+                your <code>.csv</code> file and your image files.
+            </p>
+            <Form>
+                <Form.Group className="d-flex flex-column align-items-center mb-0">
+                    <label htmlFor="file-upload-input" className="file-upload-input-label btn btn-outline-primary mb-0">
+                        <span>Upload CSV or Zip</span>
+                    </label>
+                    <Form.Control
+                        type="file"
+                        id="file-upload-input"
+                        onChange={(event) => this.handleFileUploadChange(event)}
+                        isInvalid={!fileUpload.isValid}
+                        accept=".csv,.zip"
+                        className="file-upload-input"
+                        key={fileUploadInputKey}
+                    />
+                    <Form.Control.Feedback type="invalid">{fileUpload.errorMessage}</Form.Control.Feedback>
+                </Form.Group>
+            </Form>
+        </Card.Body>)
+    }
+
+    renderUploadedFiles() {
         const { fileUpload } = this.state;
-        let formIsValid = true;
-
-        /* File Upload Validation */
-        /* A File upload is required */
-        if (fileUpload.file.size === undefined) {
-            fileUpload.isValid = false;
-            fileUpload.errorMessage = 'A file is required';
-            formIsValid = false;
-        }
+        return (<>
```

Comment:
What does returning the `<>` empty tag at the beginning do?

Reply:
When you make a JSX value it must be only one root element. You can't have:
```
return (
    <div/>
    <div/>
);
```

Originally how I would get around this was
```
return (
    <div>
        <div/>
        <div/>
    </div>
)
```

But in this use case I need `Card.Body` and `Card.Footer` without being wrapped in another element. As I learned from Ben's PR, you can actually make empty elements in react. So, by wrapping these two in an empty element I can return both of them as root level elements, basically, and avoid messing up Bootstrap

Reply:
That's nice.